api/src/pcapi/core/subscription/api.py

```diff
@@ -668,80 +805,22 @@ def get_first_registration_date(
     return min(registration_dates_when_eligible) if registration_dates_when_eligible else None
 
 
-def get_subscription_message(user: users_models.User) -> models.SubscriptionMessage | None:
+def _get_subscription_message(
+    user: users_models.User, fraud_check: fraud_models.BeneficiaryFraudCheck | None
+) -> models.SubscriptionMessage | None:
     """The subscription message is meant to help the user have information about the subscription status."""
-    if not users_api.is_eligible_for_beneficiary_upgrade(user, user.eligibility):
-        return None
-
-    next_step = get_next_subscription_step(user)
-
-    if next_step is not None and next_step not in [
-        models.SubscriptionStep.IDENTITY_CHECK,
-        models.SubscriptionStep.MAINTENANCE,
-    ]:
-        # in this case the user is not supposed to need any information and can proceed with the subscription
-        return None
-
-    if fraud_repository.has_admin_ko_review(user):
-        return subscription_messages.get_generic_ko_message(user.id)
-
-    if next_step == models.SubscriptionStep.MAINTENANCE:
+    if not get_allowed_identity_check_methods(user):
```

Comment:
Attention il y a une légère modification de logique en remontant ça. Je pense qu'on a plutôt envie de gérer ce cas dans la méthode parente, au moment où tu calcules : 
`id_check_next_step = MAINTENANCE if not not get_allowed_identity_check_methods()`

Et ce cas arrive uniquement quand le cas `if identity_check_fraud_status == models.SubscriptionItemStatus.TODO:`